src/main/java/core/basesyntax/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.suppliers;
+
+import core.basesyntax.Figure;
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_RANDOM_MULTIPLIER = 50;
+    private static final int FIGURES_COUNT = 5;
+    private static final int DEFAULT_CIRCLE_RADIUS = 10;
+    private final Random random = new Random();
+    private final Random figureRandom = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        String randomColor = colorSupplier.getRandomColor();
+        double randomDouble1 = getRandomDoubleValue();
+        double randomDouble2 = getRandomDoubleValue();
+
+        switch (figureRandom.nextInt(FIGURES_COUNT - 1)) {
+            case 0:
+                return new Circle(randomColor, randomDouble1);
+            case 1:
+                return new RightTriangle(randomColor, randomDouble1, randomDouble2);
+            case 2:
+                return new Square(randomColor, randomDouble1);
+            case 3:
+                return new Rectangle(randomColor, randomDouble1, randomDouble2);
+            default:
+                double height = getRandomDoubleValue();
+                return new IsoscelesTrapezoid(randomColor, randomDouble1, randomDouble2, height);
+        }
+    }
+
```

Comment:
consider going through the enum of shapes for example to make It more clear what cases do you have
for example
`case CIRCLE: `, not `case 3:`

Reply:
it's the optional one, like a good practice

Reply:
I agree, that is a good practice.
But it is making code harder to support (every time you add new Figure class, you should update Figures enum), isn't it?
upd: oh, wait, I have to update figures_count constant every time too.

The part in mate code style recommendations, saying that the last Figure should be returned in default branch, is also confusing me. It breaks the random if we have wrong amount of figures. I would log an Error for default branch.